azure-iot-hub/tests/test_iothub_configuration_manager.py

```diff
@@ -51,6 +53,65 @@ def iothub_configuration_manager():
     return iothub_configuration_manager
 
 
+@pytest.mark.describe("IoTHubConfigurationManager - .from_connection_string()")
+class TestFromConnectionString(object):
+    @pytest.mark.it(
+        "Creates an instance of ConnectionStringAuthentication and passes it to IotHubGatewayServiceAPIs constructor"
+    )
+    def test_connection_string_auth(self, mocker):
+        connection_string_auth_init_mock = mocker.patch.object(
+            azure.iot.hub.auth, "ConnectionStringAuthentication"
+        )
+        connection_string_auth_mock = connection_string_auth_init_mock.return_value
+        connection_string_auth_mock.__getitem__.return_value = fake_hostname
+        protocol_client_init_mock = mocker.patch.object(
+            azure.iot.hub.protocol.iot_hub_gateway_service_ap_is, "IotHubGatewayServiceAPIs"
+        )
+
+        connection_string = "HostName={hostname};DeviceId={device_id};SharedAccessKeyName={skn};SharedAccessKey={sk}".format(
+            hostname=fake_hostname,
+            device_id=fake_device_id,
+            skn=fake_shared_access_key_name,
+            sk=fake_shared_access_key,
+        )
+
+        IoTHubConfigurationManager.from_connection_string(connection_string=connection_string)
+
+        assert connection_string_auth_init_mock.call_args == mocker.call(connection_string)
+        assert protocol_client_init_mock.call_args == mocker.call(
+            connection_string_auth_mock, "https://" + connection_string_auth_mock["HostName"]
+        )
+
+
+@pytest.mark.describe("IoTHubConfigurationManager - .from_token_credential()")
+class TestFromTokenCredential(object):
```

Comment:
Like above, I don't see tests for bad inputs, i.e. invalid hostnames. Presumably there's some kind of error condition there?

Reply:
Discussed on teams